.github/workflows/test.yml

```
@@ -107,115 +90,98 @@ jobs:
             fi
           done
         shell: bash
-      - name: "[linux] Upload coverage report"
-        if: runner.os == 'Linux' && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository)
+
+  coverage:
+    runs-on: ubuntu-latest
+    permissions:
+      code-quality: write
```

actionlint (1.7.12)

[error] unknown permission scope "code-quality". all available permission scopes are "actions", "artifact-metadata", "attestations", "checks", "contents", "deployments", "discussions", "id-token", "issues", "models", "packages", "pages", "pull-requests", "repository-projects", "security-events", "statuses"

(permissions)